web-app-router/scenes/Portal/Teams/TeamId/Apps/AppId/Actions/page/createAction.tsx

```diff
@@ -0,0 +1,204 @@
+"use client";
+
+import { CloseIcon } from "@/components/Icons/CloseIcon";
+import { UserHelpNav } from "@/components/UserHelpNav";
+import Link from "next/link";
+import { useParams, usePathname } from "next/navigation";
+import { Input } from "@/components/Input";
+import * as yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Controller, useForm } from "react-hook-form";
+import { useCallback, useEffect } from "react";
+import slugify from "slugify";
+import { ApolloError } from "@apollo/client";
+import { toast } from "react-toastify";
+import { useInsertActionMutation } from "../graphql/insert-action.generated";
```

Comment:
@andy-t-wang Let's keep gql queries and generated code as close to the component where it is used as we can. 
So in this case I'd suggest to create: 
- `/CreateActionModal/graphql/<all gql and generated files>`
- `/CreateActionModal/index.tsx`

Reply:
will move the other graphql queries once I start using them